File: src/test/java/ar/edu/unlam/tallerweb1/delivery/controllers/AddFriendTest.java
package ar.edu.unlam.tallerweb1.delivery.controllers;

import ar.edu.unlam.tallerweb1.domain.FriendsGroupService;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import javax.servlet.http.HttpServletRequest;

import static org.mockito.Mockito.*;

public class AddFriendTest {

    private static final String USER = "user_1";
    private static final String FRIEND = "friend";
    private HttpServletRequest request;
    private AddFriend controller;
    private FriendsGroupService friendsGroupService;
    @BeforeEach
    public void init() {
        request = mock(HttpServletRequest.class);
        when(request.getParameter("user")).thenReturn(USER);
        when(request.getParameter("friend")).thenReturn(FRIEND);
        friendsGroupService = mock(FriendsGroupService.class);

        controller = new AddFriend(friendsGroupService);
    }
    @Test
    public void addFriendToUserFriendGroup(){
        whenAddFriendToFriendsGroupOf(USER, FRIEND);
        thenAddFriendToGroup(FRIEND);
    }

    private void whenAddFriendToFriendsGroupOf(String user, String friend) {
        controller.invoke(request);
    }

    private void thenAddFriendToGroup(String friend) {
        verify(friendsGroupService, times(1)).addFriendsToGroup(USER, friend);
    }
}

File: src/test/java/ar/edu/unlam/tallerweb1/delivery/controllers/AddFriendTest.kt
package ar.edu.unlam.tallerweb1.delivery.controllers

import ar.edu.unlam.tallerweb1.domain.usecases.AddFriendsToGroup
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.kotlin.*
import javax.servlet.http.HttpServletRequest

class AddFriendTest {
    private val USER = "user_1"
    private val FRIEND = "friend"
    private lateinit var controller: AddFriend
    private lateinit  var request: HttpServletRequest
    private lateinit var addFriendsToGroup: AddFriendsToGroup

    @BeforeEach
    fun init() {
        request = mock()
        whenever(request.getParameter("user"))   doReturn USER
        whenever(request.getParameter("friend")) doReturn FRIEND
        addFriendsToGroup = mock()
        controller = AddFriend(addFriendsToGroup)
    }

    @Test
    fun addFriendToUserFriendGroup() {
        whenAddFriendToFriendsGroupOf(USER, FRIEND)
        thenAddFriendToGroup(FRIEND)
    }

    private fun whenAddFriendToFriendsGroupOf(user: String, friend: String) = controller(request)

    private fun thenAddFriendToGroup(friend: String) =
        verify(addFriendsToGroup, times(1)).invoke(USER, friend)
}
